PDP11/pdp11_rr.c

```diff
@@ -1083,7 +1178,7 @@ static t_stat rr_reset (DEVICE *dptr)
 {
     int32 i;
 
-    /* some sanity check first */
+    /* compile-time sanity check first */

```

Comment:
That doesn't seem to be a compile time test.  Do we have STATIC_ASSERT or something like that?  It would be nice to use that when possible.

Reply:
> That doesn't seem to be a compile time test.

It is.  The condition is a constant expression known at the time of compilation.  So `assert()` would have been evaluated by the compiler, and there would (most likely) be an unconditional call for the assert handler if the condition was false.  That's what I mean.  At any rate, `assert()` is not compiled at all when the simulators are built without `DEBUG=1` as make's argument.

Reply:
Sure, so it's a run time call to assert with a constant (evaluated at compile time) argument.  
There is such a thing as "compile time assert" but it's a different thing entirely, and very useful when available: it's a macro, or compiler built-in operation, that is completely handled at compile time.  If the check succeeds, no code is generated at all, ever.  If the check fails, you get a compile error.  So the failure shows up at build time, not run time as an assert does.
There are various ways of doing this; one relies on the fact that arrays with negative bounds are invalid.  So something like
```
#define static_assert(cond) { char assert ## __LINE__[-(!(cond))]; }
```
give or take a few details.